src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java

```diff
@@ -0,0 +1,96 @@
+package uk.gov.hmcts.reform.idam.web.strategic;
+
+import com.google.common.collect.ImmutableMap;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.idam.api.external.model.ActionMap;
+import uk.gov.hmcts.reform.idam.api.external.model.EvaluatePoliciesRequest;
+import uk.gov.hmcts.reform.idam.api.external.model.EvaluatePoliciesResponse;
+import uk.gov.hmcts.reform.idam.api.external.model.EvaluatePoliciesResponseInner;
+import uk.gov.hmcts.reform.idam.api.external.model.Subject;
+import uk.gov.hmcts.reform.idam.web.config.properties.ConfigurationProperties;
+
+import java.util.Collections;
+
+import static java.util.Collections.singletonList;
+import static java.util.Optional.ofNullable;
+
+@Slf4j
+@Service
+public class PolicyService {
+
+    public static final String ERROR_POLICY_CHECK_EXCEPTION = "Policy check exception.";
+
+    private final RestTemplate restTemplate;
+
+    private final ConfigurationProperties configurationProperties;
+
+    @Autowired
+    public PolicyService(RestTemplate restTemplate, ConfigurationProperties configurationProperties) {
+        this.restTemplate = restTemplate;
+        this.configurationProperties = configurationProperties;
+    }
+
+    /**
+     * @should return true when all actions return true
+     * @should return false when any action returns false
+     * @should throw exception when response is not successful
+     * @should return true when no actions are returned
+     * @should return true when actions is null
+     */
+    public boolean evaluatePoliciesForUser(final String uri, final String cookie, final String ipAddress) {
```

Comment:
Almost everything in this method is about making the rest call, and then the evaluation of the response is in another method. I wonder if it would make more sense to put all of the rest stuff into a method like `public EvaluatePoliciesResponse doEvaluatePolicies(...)` and then this method just calls that, and then checks the response. If `doEvaluatePolicies` moved to a separate class then it would be easier to mock for testing, but then it's starting to look like feign, and I guess you've already written the tests, so probably not worth it.